src/content/blog/astro-570.mdx

```diff
@@ -0,0 +1,235 @@
+authors:
+  - matt
+  - florian
+  - ema
+  - nate
+coverImage: '/src/content/blog/_images/astro-570/blog-post-5-7.webp'
+socialImage: '/src/content/blog/_images/astro-570/og-astro-5-7.webp'
+lang: 'en'
+---
+
+import BlogContentImage from '/src/components/BlogContentImage.astro';
+import Note from '~/components/Note.astro';
+import Mention from '/src/components/Mention.astro';
+
+**Astro 5.7 has a basketload of treats, including stable Sessions and SVG components and a new Experimental Fonts API.**
+
+🐣 Full of the joys of northern-hemisphere spring, Astro 5.7 is here with goodies galore:
+
+- [**Experimental Fonts API**](#experimental-fonts-api)
+- [**Sessions API**](#sessions-api)
+- [**SVG Components**](#svg-components)
+- [**Config Imports**](#config-imports)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental Fonts API
+
+...
+
+## Sessions API
+
+The Astro Sessions API is now stable and ready for production!
+
+Sessions let you securely store data tied to a specific user as they navigate between pages on your site. Unlike cookies, sessions are stored on the server, so you can store larger amounts of data without worrying about size limits or security issues. They are useful for storing things like user data, shopping carts, and form state, and they work without any client-side JavaScript:
+
+
+```astro title="src/components/CartButton.astro" {2}
+---
+const cart = await Astro.session.get('cart');
+export const prerender = false;
+---
+
+<a href="/checkout">🛒 {cart?.length ?? 0} items</a>
+```
+
+They can be used in Astro components, pages, API endpoints, actions, and middleware. The API is the same for all of them:
+
+```ts title="src/actions/addToCart.ts" {9,10}
+import { defineAction } from "astro:actions";
+import { z } from "astro:schema";
+export const server = {
+  addToCart: defineAction({
+    input: z.object({
+      product: z.string(),
+    }),
+    async handler({ product }, { session }) {
+      const cart = await session.get('cart');
+      session.set('cart', [...cart, product]);
+    },
+  }),
+};
+```
+
+### Type-safety
+
+Of course, as this is Astro, your sessions can be type-safe too. Define an interface and you will get type-checking and auto-completion for all of your session data:
+
+
+```ts title="src/env.d.ts"
+declare namespace App {
+  interface SessionData {
+    user: {
+      id: string;
+      name: string;
+    };
+    cart: string[];
+  }
+}
+```
+
+### Cookie-less sessions
+
+The session ID is automatically tracked using a cookie, but you can also use them without cookies by using the `session.load()` method. This is useful for things like APIs, where you might want to use a custom header or params to track the session ID:
```

Comment:
```suggestion
The session ID is automatically tracked using a cookie, but you can also refer to sessions without cookies by using the `session.load()` method. This is useful for things like APIs, where you might want to use a custom header or params to track the session ID:
```
refer to sessions?
call sessions?
load sessions?
access a session?

Just trying to avoid a double "use"/"using" here and being a bit more precise. Is there anything here that works?

Alternatively, something like:

> but you can instead use the `session.load()` method to (description of action here: refer to/access/identify...

Reply:
```suggestion
The session ID is automatically tracked using a cookie, but you can also use the `session.load()` method to access sessions without cookies . This is useful for things like APIs, where you might want to use a custom header or params to track the session ID:
```